torch/nn/functional.py

```diff
@@ -4916,11 +4916,11 @@ def _in_projection(
 
     # Efficient implementation equivalent to the following:
     scale_factor = 1 / math.sqrt(Q.size(-1)) if scale is None else scale
-    attn_mask = torch.ones(L, S, dtype=torch.bool).tril(diagonal=0) if is_causal else attn_mask
-    attn_mask = attn_mask.masked_fill(not attn_mask, -float('inf')) if attn_mask.dtype==torch.bool else attn_mask
+    attn_mask = torch.ones(L, S, dtype=torch.bool).tril(diagonal=0) if is_causal else torch.zeros(L, S, dtype=Q.dtype)
+    attn_mask = attn_mask.masked_fill(~attn_mask, -float('inf')) if attn_mask.dtype==torch.bool else attn_mask
```

Comment:
same question here: can attn_mask be filled with float values if it's bool dtype?

It would be nice to have this sample run automatically to check it's correct